doc/src/guides/contributing/deprecations.md

```diff
@@ -0,0 +1,388 @@
+(deprecation-policy)=
+# Deprecation Policy
+
+This page outlines SymPy's police on doing deprecations, and describes the
+steps developers should take to properly deprecate code.
+
+## What is a deprecation?
+
+A deprecation is a way to make backwards incompatible changes in a way that
+allows users to update their code. Deprecated code continues to work as it
+used to, but whenever someone uses it, it prints a warning to the screen
+indicating that it will be removed in a future version of SymPy, and
+indicating what the user should be using instead.
+
+This allows users a chance to update their code without it completely
+breaking. It also gives SymPy an opportunity to give users an informative
+message on how to update their code, rather than making their code simply
+error or start giving wrong answers.
+
+## When to deprecate and when not to
+
+Sometimes the API of SymPy must change in an incompatible way. We try to avoid
```

Comment:
Suggested change:

"We should, in general, avoid this,"

Below it is emphasized in multiple places why changing API is not good, so I think opening with a bit stronger statement is best.

Reply:
I did have something written about this but it was lower down. I've moved it to the top of the file and given it its own heading.